File: 3249-minimum-number-of-operations-to-make-array-xor-equal-to-k/3249-minimum-number-of-operations-to-make-array-xor-equal-to-k.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int total_min = 0;
        for(auto it : nums)
        {
            total_min ^= it; 
        }

        int ans = total_min ^ k ;

        return __builtin_popcount(ans);
    }
};

File: 3249-minimum-number-of-operations-to-make-array-xor-equal-to-k/3249-minimum-number-of-operations-to-make-array-xor-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int total_min = 0;
        for(int it : nums)
        {
            total_min ^= it;
        }

        int ans = total_min ^ k;

        return Integer.bitCount(ans);
    }
}